### PR TITLE
make Oracle compilers happy panda again

### PR DESCRIPTION
**Pull request by ggouaillardet**
Thanks Siegmar Gross for the report

### CODE REVIEW

**Comment by ggouaillardet** [comment]
@karasevb that fixes the issue reported at https://mail-archive.com/users@lists.open-mpi.org/msg30625.html
the root cause is very likely a workaround in Oracle compilers
(a compiler should never crash !)
i revamped the macros (fortunatly, it is not un-elegant) and the compiler stopped crashing ...

File: src/dstore/pmix_esh.c [hunk 1]
```diff
@@ -1092,10 +1084,11 @@ int _esh_store(const char *nspace, pmix_rank_t rank, pmix_kval_t *kv)
     }
 
     /* set exclusive lock */
-    if (PMIX_SUCCESS != (rc = _ESH_WRLOCK(_ESH_SESSION_lock(ns_map->tbl_idx)))) {
+    if (PMIX_SUCCESS != (rc = _ESH_WRLOCK(PMIX_VALUE_ARRAY_GET_BASE(_session_array, session_t)[ns_map->tbl_idx].rwlock))) {
         PMIX_ERROR_LOG(rc);
         return rc;
     }
+#if 1
```
Comment (rhc54) [inline src/dstore/pmix_esh.c]: What does this do?

Comment (ggouaillardet) [inline src/dstore/pmix_esh.c]: my bad, this is debug leftover ...
will fix shortly

**Comment by ggouaillardet** [comment]
@rhc54 i removed the debug and an extra useless change

**Pull request merged**